master/internal/command/command.go

```diff
@@ -327,20 +327,29 @@ func (c *command) Receive(ctx *actor.Context) error {
 		ctx.Respond(c.setWeight(ctx, msg.Weight))
 
 	case job.SetGroupPriority:
-		ctx.Respond(c.setPriority(ctx, msg.Priority))
+		ctx.Respond(c.setPriority(ctx, msg.Priority, msg.Handler))
+
+	case job.RegisterJobPosition:
+		err := c.db.UpdateJobPosition(msg.JobID, msg.JobPosition)
+		if err != nil {
+			ctx.Log().WithError(err).Errorf("persisting position for job %s failed", msg.JobID)
+		}
 
 	default:
 		return actor.ErrUnexpectedMessage(ctx)
 	}
 	return nil
 }
 
-func (c *command) setPriority(ctx *actor.Context, priority int) error {
+func (c *command) setPriority(ctx *actor.Context, priority int, handler *actor.Ref) error {
 	if sproto.UseK8sRM(ctx.Self().System()) {
 		return fmt.Errorf("setting priority for job type %s in kubernetes is not supported",
 			c.jobType)
 	}
 	c.Config.Resources.Priority = &priority
+	if handler != nil {
+		return nil
+	}
```

Comment:
this is very confusing to me. if the handler is not nil, we return and don't actually set prio? but then it seems like we don't use handler at all to send the message (and i dont understand why we dont send it in all cases)

Reply:
mb this was a late night debugging hotfix that I forgot to mark. The problem is just that you can enter a loop and hang in certain cases. Most of the time, it's the group handler sending this message to the RP, but in the case of moving a job, it becomes RP -> group handler -> RP. So the fix was just to check for a handler. You're right that it really should be a check for what the handler is and where it comes from.

Reply:
this is still funny, were we going to change this @eecsliu ? just checking.

Reply:
why are commands not handling `NotifyRMPriorityChange`?